src/app/screens/Home/index.tsx

```diff
@@ -86,19 +86,35 @@ function Home() {
 
   const loadPayments = useCallback(async () => {
     try {
-      const result = await api.getPayments({ limit: 10 });
-      for await (const payment of result.payments) {
-        const totalAmountFiat = settings.showFiat
-          ? await getFiatValue(payment.totalAmount)
-          : "";
-        payment.totalAmountFiat = totalAmountFiat;
+      if (allowance) {
+        const payments: Transaction[] = allowance.payments.map((payment) => ({
```

Comment:
I would change this and do `api.getPayments` call with a host (we don't set an allowance Id).

then these two cases do not need to be handled differently and in this screen they are exactly the same. 

see my comment with that diff